### DISCUSSION TITLE
Touchscreen - Double Clicking on Nodes Not Working

### DISCUSSION THREAD

**Discussion by spaceymonk** [xyflow/xyflow#2210] [Q&A] [answered]
Hello everyone,
I did not want to open an issue about this, maybe I am doing it wrong. The thing is, I can double click on a node on the desktop with no problem. But on mobile, clicking on nodes only moves them around. What can I do to trigger the `onNodeDoubleClick` event on touchscreen devices? Thank you.

**Comment by bcakmakoglu** [comment, marked as answer]
Can you provide an example? 
I just used [this](https://codesandbox.io/s/sandpack-project-forked-k6sj1f?file=/App.js) sandbox on my phone and the console logs "dbl click" any time I tap a node twice - so everything seems to be working as expected.

**Reply by spaceymonk** [reply]
The issue I am having in this [file](https://github.com/spaceymonk/flower-app/blob/8bf974ebb93b62ef6d0528e377a2559d7af7e8e1/src/components/board/Board.tsx#L89). You can visit [the project website](https://web.itu.edu.tr/~ozkanbe19/flower-app/) and try to create a block (➕ icon on the top-right corner of the toolbar and then select a block from the opened sidebar). If you are on the desktop you can double click on the created node to open a modal, but this is not possible on mobile.

Moreover, I have tried to print node id on both console and DOM when a double click occurred, but no luck on mobile.

Edit: Your example works on my phone too.

**Reply by bcakmakoglu** [reply]
Can you try updating `react-flow-renderer` to the latest version or is that an issue for you currently?
I believe there were some fixes for touch-clicks in the recent versions (not sure which exact one) and I saw you're using `^10.0.0` (current version is `10.3.6`)

**Reply by spaceymonk** [reply]
Thank you! Upgrading the version solved the issue.

Edit: I was having a TS complaint but I resolved it. Apparently, the parameters of the Node Components changed also.
